compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java

```diff
@@ -119,6 +119,10 @@ private void writeType(ByteBuf buf, BIRTypeWriter typeWriter, BIRTypeDefinition
         // Visibility
         buf.writeByte(typeDef.visibility.value());
         typeDef.type.accept(typeWriter);
+
+        if (typeDef.attachedFuncs != null) {
```

Comment:
do we need this null check?
(BTW I would like this to be written as top-level nodes, however, I understand why you did serialize this at type definition level, anyway would like to see the possibility of serializing this as top-level nodes)

Reply:
so looks like we are setting this to null somewhere else. check - https://github.com/ballerina-platform/ballerina-lang/pull/14675#pullrequestreview-218690870